src/plugins/vis_type_timeseries/public/application/components/aggs/top_hit.js

```diff
@@ -110,7 +110,7 @@ const TopHitAggUi = (props) => {
     PANEL_TYPES.METRIC,
     PANEL_TYPES.MARKDOWN,
   ].includes(panel.type)
-    ? [KBN_FIELD_TYPES.NUMBER, KBN_FIELD_TYPES.STRING]
+    ? [KBN_FIELD_TYPES.NUMBER, KBN_FIELD_TYPES.STRING, KBN_FIELD_TYPES.DATE]
```

Comment:
You are right here. 

- For `TimeSeries/Gauge` we support only `KBN_FIELD_TYPES.NUMBER` field type. 
- For `Table/Metric/Markdown` - `KBN_FIELD_TYPES.NUMBER, KBN_FIELD_TYPES.STRING, KBN_FIELD_TYPES.DATE`

The main problem in your PR and code which was implemented before your changes that there are no correct `Aggregation` function for `String` and `Date`. 

Currenly it's undefined but I prefer to do it like in classical visualizations with Concatenate option 

![image](https://user-images.githubusercontent.com/20072247/123630142-d5599080-d81d-11eb-84f9-972e635baabf.png)

Screen if I select Date field
![image](https://user-images.githubusercontent.com/20072247/123630376-1f427680-d81e-11eb-99dd-c0a10437d5a1.png)